### ISSUE TITLE
UPDATE EDGE command does not check the graph consistency properly

### ISSUE THREAD

**Issue by xscratt** [orientechnologies/orientdb#6378]
## Expected behavior and actual behavior

UPDATE EDGE command does not check the graph consistency in proper way when change `in`/`out` properties of an edge.

**Actual behavior**: new vertex' `in`/`out` is updating, but the vertex that was previously assigned to the edge still contain info about the edge.
**Expected behavior**: `in`/`out` of both new and old vertices changed properly - the edge reference removed from the old vertex, and added to the new one.

Checked on OrientDB 2.2.0 and 2.2.3 and reproduced on both.

## Steps to reproduce the problem

Small example that illustrates this behaviour.
```
CREATE VERTEX V SET label='vertex 1'
CREATE VERTEX V SET label='vertex 2'
CREATE VERTEX V SET label='vertex 3'
CREATE EDGE E FROM #9:0 to #10:0

SELECT FROM V

+-------+----------+--------+----------+-------+-------+
| @rid  | @version | @class | label    | IN    | OUT   |
+-------+----------+--------+----------+-------+-------+
|  #9:0 | 2        | V      | vertex 1 |       | #13:0 |
| #10:0 | 2        | V      | vertex 2 | #13:0 |       |
| #11:0 | 1        | V      | vertex 3 |       |       |
+-------+----------+--------+----------+-------+-------+

SELECT FROM E

+-------+----------+--------+-------+-------+
| @rid  | @version | @class | OUT   | IN    |
+-------+----------+--------+-------+-------+
| #13:0 | 1        | E      | #9:0  | #10:0 |
+-------+----------+--------+-------+-------+
```
Change the edge endpoint to another vertex:
```
UPDATE EDGE #13:0 SET in=#11:0
SELECT FROM E

+-------+----------+--------+-------+-------+
| @rid  | @version | @class | OUT   | IN    |
+-------+----------+--------+-------+-------+
| #13:0 | 2        | E      | #9:0  | #11:0 |
+-------+----------+--------+-------+-------+

SELECT FROM V

+-------+----------+--------+----------+-------+-------+
| @rid  | @version | @class | label    | IN    | OUT   |
+-------+----------+--------+----------+-------+-------+
|  #9:0 | 2        | V      | vertex 1 |       | #13:0 |
| #10:0 | 2        | V      | vertex 2 | #13:0 |       |
| #11:0 | 2        | V      | vertex 3 | #13:0 |       |
+-------+----------+--------+----------+-------+-------+
```
Values of `in` for both edge `#13:0` and vertex `#11:0` has been changed properly (and its versions also has been increased), but the vertex `#10:0` still contain info about the edge (and has unchanged version). If change `out` property of an edge will get the same result.

### Runninng Mode
- [ ] Embedded, using PLOCAL access mode
- [ ] Embedded, using MEMORY access mode
- [x] Remote (with OrientDB studio)

### Misc
- [ ] I have a distributed setup with multiple servers. How many? 
- [ ] I'm using the Enterprise Edition

### OrientDB Version
- [ ] v2.0.x
- [ ] v2.1.x
- [x] v2.2.x - checked on 2.2.0 and 2.2.3

### Operating System
- [x] Linux, image from [dockerhub](https://hub.docker.com/_/orientdb/)
- [ ] MacOSX
- [ ] Windows
- [ ] Other Unix
- [ ] Other, name? 

### Java Version
- [ ] 6
- [ ] 7
- [x] 8

**Comment by luigidellaquila** [comment]
Hi @xscratt 

this seems to be a regression, I'm checking it now

Thanks

Luigi

**Comment by luigidellaquila** [comment]
Hi @xscratt 

the problem seems to happen only with E class (not with subclasses). 
I should have a fix in a few hours, I'll let you know

Thanks

Luigi

**Comment by xscratt** [comment]
Hi @luigidellaquila

Thanks for quick response!
It also reproduced for subclasses of E class (just checked on OrientDB 2.2.3):
```
CREATE CLASS Connected EXTENDS E
CREATE VERTEX V SET label='vertex 1'
CREATE VERTEX V SET label='vertex 2'
CREATE VERTEX V SET label='vertex 3'
CREATE EDGE Connected FROM #9:0 to #10:0

SELECT FROM V
+-------+----------+--------+----------+-------+-------+
| @rid  | @version | @class | label    | IN    | OUT   |
+-------+----------+--------+----------+-------+-------+
|  #9:0 | 2        | V      | vertex 1 |       | #21:0 |
| #10:0 | 2        | V      | vertex 2 | #21:0 |       |
| #11:0 | 1        | V      | vertex 3 |       |       |
+-------+----------+--------+----------+-------+-------+

SELECT FROM E
+-------+----------+-----------+-------+-------+
| @rid  | @version | @class    | OUT   | IN    |
+-------+----------+-----------+-------+-------+
| #21:0 | 1        | Connected | #9:0  | #10:0 |
+-------+----------+-----------+-------+-------+

UPDATE EDGE #21:0 SET in=#11:0

SELECT FROM V
+-------+----------+--------+----------+-------+-------+
| @rid  | @version | @class | label    | IN    | OUT   |
+-------+----------+--------+----------+-------+-------+
|  #9:0 | 2        | V      | vertex 1 |       | #21:0 |
| #10:0 | 2        | V      | vertex 2 | #21:0 |       |
| #11:0 | 2        | V      | vertex 3 | #21:0 |       |
+-------+----------+--------+----------+-------+-------+

SELECT FROM E
+-------+----------+-----------+-------+-------+
| @rid  | @version | @class    | OUT   | IN    |
+-------+----------+-----------+-------+-------+
| #21:0 | 2        | Connected | #9:0  | #11:0 |
+-------+----------+-----------+-------+-------+
```
And the same result if create edge of E class when E has subclasses. But I believe that the fix in E class will solve the problem.

Thank you

**Comment by luigidellaquila** [comment]
Hi @xscratt 

You're right, the problem also happened with other edge classes.

I just pushed a fix on 2.2.x and develop, it will be released with 2.2.4

Thanks

Luigi

**Comment by xscratt** [comment]
Thanks!
